Sources/MongoSwift/BSON/Document.swift

```diff
@@ -243,26 +261,17 @@ public struct Document: ExpressibleByDictionaryLiteral, ExpressibleByArrayLitera
         // otherwise, it's a new key
         } else {
             self.copyStorageIfRequired()
-
-            if let value = newValue {
-                try value.encode(to: self.storage, forKey: key)
-            } else {
-                guard bson_append_null(self.data, key, Int32(key.count)) else {
-                    throw MongoError.bsonEncodeError(message: "Failed to set the value for key \(key) to null")
-                }
-            }
+            try newValue.encode(to: self.storage, forKey: key)
         }
     }
 
-    /// Retrieves the value associated with for as a `BSONValue?`, which can be nil.
+    /// Retrieves the value associated with for as a `BSONValue?`, which can be nil if the key does not exist in the
```

Comment:
I might suggest adding backticks around "for" so this reads more clearly (as referring to the `for` argument).

Reply:
Great idea! Done.